mobile/src/androidTest/java/ch/epfl/sdp/mobile/test/state/StatefulGameScreenTest.kt

```diff
@@ -675,4 +705,71 @@ class StatefulGameScreenTest {
     robot.onNodeWithContentDescription(robot.strings.boardPieceQueen).performClick()
     robot.onNodeWithLocalizedText { robot.strings.gamePromoteConfirm }.assertIsNotEnabled()
   }
+
+  @Test
+  fun given_successfulRecognizer_when_clicksListening_then_displaysRecognitionResults() {
+    // This will be fail once we want to move the pieces instead.
+    val robot =
+        emptyGameAgainstOneselfRobot(
+            recognizer = SuccessfulSpeechRecognizerFactory,
+            audioPermission = GrantedPermissionState,
+        )
+    robot.onNodeWithLocalizedContentDescription { gameMicOffContentDescription }.performClick()
+    robot.onNodeWithText(SuccessfulSpeechRecognizer.Results[0]).assertExists()
+  }
+
+  @Test
+  fun given_failingRecognizer_when_clicksListening_then_displaysFailedRecognitionResults() {
+    // This will be fail once we want to move the pieces instead.
+    val robot =
+        emptyGameAgainstOneselfRobot(
+            recognizer = FailingSpeechRecognizerFactory,
+            audioPermission = GrantedPermissionState,
+        )
+    robot.onNodeWithLocalizedContentDescription { gameMicOffContentDescription }.performClick()
+    robot.onNodeWithText("Internal failure").assertExists()
+  }
+
+  @Test
+  fun given_noPermission_when_clicksListening_then_requestsPermission() {
+    val permission = MissingPermissionState()
+    val robot =
+        emptyGameAgainstOneselfRobot(
+            recognizer = SuspendingSpeechRecognizerFactory,
+            audioPermission = permission,
+        )
+    robot.onNodeWithLocalizedContentDescription { gameMicOffContentDescription }.performClick()
+    assertThat(permission.permissionRequested).isTrue()
+  }
+
+  @Test
+  fun given_suspendingRecognizer_when_clickingListeningTwice_then_cancelsRecognition() {
+    val robot =
+        emptyGameAgainstOneselfRobot(
+            recognizer = SuspendingSpeechRecognizerFactory,
+            audioPermission = GrantedPermissionState,
+        )
+    robot.onNodeWithLocalizedContentDescription { gameMicOffContentDescription }.performClick()
+    robot.onNodeWithLocalizedText { gameListening }.performClick()
+    robot.onNodeWithLocalizedContentDescription { gameMicOffContentDescription }.assertExists()
+  }
+}
+
+private object GrantedPermissionState : PermissionState {
+  override val hasPermission = true
+  override val permission = RECORD_AUDIO
+  override val permissionRequested = true
+  override val shouldShowRationale = false
+  override fun launchPermissionRequest() = Unit
+}
+
+private class MissingPermissionState : PermissionState {
```

Comment:
Why is this a class instead pf an object ?

Reply:
To make sure the instance is not shared across multiple tests.